xonsh/lib/itertools.py

```diff
@@ -0,0 +1,33 @@
+def as_iterable(iterable_or_scalar):
+    """Utility for converting an object to an iterable.
+   Parameters
+   ----------
+   iterable_or_scalar : anything
+   Returns
+   -------
+   l : iterable
+       If `obj` was None, return the empty tuple.
+       If `obj` was not iterable returns a 1-tuple containing `obj`.
+       Otherwise return `obj`
+   Notes
+   -----
+   Although both string types and dictionaries are iterable in Python, we are treating them as not iterable in this
```

Comment:
Seems like the `Notes` here don't match the implementation anymore, otherwise, this looks good to me. Sorry for the delay in getting to it!